### ISSUE TITLE
dgraph server 'segmentation violation' on startup

### ISSUE THREAD

**Issue by lsnider** [dgraph-io/dgraph#2117]
- What version of Dgraph are you using?
1.0.3

- Have you tried reproducing the issue with latest release?
Yes.  We were on 1.0.2-dev.  Upgrade to 1.0.3 and tried again. 

- What is the hardware spec (RAM, OS)?
12GB RAM, Ubuntu 16.04.3 LTS

- Steps to reproduce the issue (command/config used to run Dgraph).
We have three host machines setup each with a zero and a server running.  Here are the commands we are running for each:
```
dgraph0
/usr/local/bin/dgraph server --bindall=true --idx=459 --memory_mb=2048 --zero=192.168.206.84:5080 --my=192.168.206.84:7080
/usr/local/bin/dgraph zero --bindall=true --idx=459 --replicas=3 --my=192.168.206.84:5080


dgraph1
/usr/local/bin/dgraph server --bindall=true --idx=460 --memory_mb=2048 --zero=localhost:5080 --my=192.168.171.220:7080
/usr/local/bin/dgraph zero --bindall=true --idx=460 --peer=192.168.200.88:5080 --replicas=3 --my=192.168.171.220:5080


dgraph2
/usr/local/bin/dgraph server --bindall=true --idx=461 --memory_mb=2048 --zero=192.168.206.84:5080 --my=192.168.200.88:7080
/usr/local/bin/dgraph zero --bindall=true --idx=461 --peer=192.168.206.84:5080 --replicas=3 --my=192.168.200.88:5080
```

While loading data in the cluster, the cluster went down.  We ran it by the group on slack who suggested it felt like one of the zero instances was down.  It may have been but all zero and server instances were running.  We verified connectivity to all ip/port combos from each host machines.

We were on 4GB of ram and upped the VMs to 12GB as we saw the server instances starting to swap even though we had them set to `--memory_mb=2048`.  

- Expected behaviour and actual result.
On the reboot, the zero instances started ok but all three server instances are having issues.  Here is what we are seeing for each:

dgraph0
```
/usr/local/bin/dgraph server --bindall=true --idx=459 --memory_mb=8192 --zero=192.168.206.84:5080 --my=192.168.206.84:7080
2018/02/12 20:26:11 gRPC server started.  Listening on port 9080
2018/02/12 20:26:11 HTTP server started.  Listening on port 8080
2018/02/12 20:26:11 groups.go:86: Current Raft Id: 459
2018/02/12 20:26:11 worker.go:99: Worker listening at address: [::]:7080
2018/02/12 20:26:11 pool.go:118: == CONNECT ==> Setting 192.168.206.84:5080
2018/02/12 20:26:11 groups.go:109: Connected to group zero. Connection state: member:<id:459 addr:"192.168.206.84:7080" > state:<counter:879041 groups:<key:1 value:<members:<key:459 value:<id:459 group_id:1 addr:"192.168.206.84:7080" last_update:1518465931 > > members:<key:460 value:<id:460 group_id:1 addr:"192.168.171.220:7080" > > members:<key:461 value:<id:461 group_id:1 addr:"192.168.200.88:7080" leader:true last_update:1518465993 > > tablets:<key:"_predicate_" value:<group_id:1 predicate:"_predicate_" space:56836755 > > tablets:<key:"dummy" value:<group_id:1 predicate:"dummy" space:17 > > tablets:<key:"has_connection" value:<group_id:1 predicate:"has_connection" space:123985764 > > tablets:<key:"name" value:<group_id:1 predicate:"name" space:38151179 > > tablets:<key:"person_id" value:<group_id:1 predicate:"person_id" space:54935649 > > > > zeros:<key:459 value:<id:459 addr:"192.168.206.84:5080" leader:true > > zeros:<key:460 value:<id:460 addr:"192.168.171.220:5080" > > zeros:<key:461 value:<id:461 addr:"192.168.200.88:5080" > > maxLeaseId:290000 maxTxnTs:4880000 >
2018/02/12 20:26:11 pool.go:168: Echo error from 192.168.171.220:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
2018/02/12 20:26:11 pool.go:118: == CONNECT ==> Setting 192.168.171.220:7080
2018/02/12 20:26:11 pool.go:168: Echo error from 192.168.200.88:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
2018/02/12 20:26:11 pool.go:118: == CONNECT ==> Setting 192.168.200.88:7080
2018/02/12 20:26:11 pool.go:118: == CONNECT ==> Setting 192.168.200.88:5080
2018/02/12 20:26:11 pool.go:118: == CONNECT ==> Setting 192.168.171.220:5080
2018/02/12 20:26:11 draft.go:140: Node ID: 459 with GroupID: 1
2018/02/12 20:26:12 node.go:231: Found Snapshot, Metadata: {ConfState:{Nodes:[459 460 461] XXX_unrecognized:[]} Index:559475 Term:38 XXX_unrecognized:[]}
2018/02/12 20:26:12 node.go:246: Found hardstate: {Term:76 Vote:459 Commit:1756177 XXX_unrecognized:[]}
2018/02/12 20:26:16 node.go:258: Group 1 found 1196702 entries
2018/02/12 20:26:16 draft.go:661: Restarting node for group: 1
2018/02/12 20:26:16 raft.go:567: INFO: 1cb became follower at term 76
2018/02/12 20:26:16 raft.go:315: INFO: newRaft 1cb [peers: [1cb,1cc,1cd], term: 76, commit: 1756177, applied: 559475, lastindex: 1756177, lastterm: 38]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x98 pc=0x1000ed6]

goroutine 1160 [running]:
github.com/dgraph-io/dgraph/posting.getNew(0xc480583120, 0x17, 0x17, 0xc4201580f8, 0x0, 0x0, 0x0)
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/posting/mvcc.go:470 +0x286
github.com/dgraph-io/dgraph/posting.Get(0xc480583120, 0x17, 0x17, 0xc480583120)
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/posting/lists.go:253 +0xca
github.com/dgraph-io/dgraph/worker.runMutation(0x1a18000, 0xc4269bab10, 0xc4791ce600, 0xc4791c04b0, 0x124e480, 0xc479182240)
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/worker/mutation.go:80 +0x231
github.com/dgraph-io/dgraph/worker.(*node).processMutation(0xc4200118f0, 0xc4791ca4e0, 0x1, 0x0)
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/worker/draft.go:279 +0x154
github.com/dgraph-io/dgraph/worker.(*scheduler).processTasks(0xc47c2577c0)
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/worker/scheduler.go:65 +0x6b
created by github.com/dgraph-io/dgraph/worker.(*scheduler).init
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/worker/scheduler.go:56 +0xe9
```

dgraph1
```
/usr/local/bin/dgraph server --bindall=true --idx=460 --memory_mb=8192 --zero=192.168.171.220:5080 --my=192.168.171.220:7080
2018/02/12 20:27:33 Assert failed
github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger/y.AssertTrue
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger/y/error.go:55
github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger.replayFunction.func2
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger/db.go:138
github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger.(*valueLog).iterate
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger/value.go:270
github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger.(*valueLog).Replay
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger/value.go:660
github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger.Open
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger/db.go:295
github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger.OpenManaged
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger/managed_db.go:40
github.com/dgraph-io/dgraph/edgraph.(*ServerState).initStorage
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/edgraph/server.go:142
github.com/dgraph-io/dgraph/edgraph.InitServerState
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/edgraph/server.go:80
github.com/dgraph-io/dgraph/dgraph/cmd/server.run
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/dgraph/cmd/server/run.go:306
github.com/dgraph-io/dgraph/dgraph/cmd/server.init.0.func1
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/dgraph/cmd/server/run.go:63
github.com/dgraph-io/dgraph/vendor/github.com/spf13/cobra.(*Command).execute
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/vendor/github.com/spf13/cobra/command.go:702
github.com/dgraph-io/dgraph/vendor/github.com/spf13/cobra.(*Command).ExecuteC
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/vendor/github.com/spf13/cobra/command.go:783
github.com/dgraph-io/dgraph/vendor/github.com/spf13/cobra.(*Command).Execute
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/vendor/github.com/spf13/cobra/command.go:736
github.com/dgraph-io/dgraph/dgraph/cmd.Execute
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/dgraph/cmd/root.go:52
main.main
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/dgraph/main.go:34
runtime.main
        /home/travis/.gimme/versions/go1.9.2.linux.amd64/src/runtime/proc.go:195
runtime.goexit
        /home/travis/.gimme/versions/go1.9.2.linux.amd64/src/runtime/asm_amd64.s:2337
```

dgraph2
```
root@dgraph2:/opt/dgraph# /usr/local/bin/dgraph server --bindall=true --idx=461 --memory_mb=8192 --zero=192.168.206.84:5080 --my=192.168.200.88:7080
2018/02/12 20:26:43 gRPC server started.  Listening on port 9080
2018/02/12 20:26:43 HTTP server started.  Listening on port 8080
2018/02/12 20:26:43 groups.go:86: Current Raft Id: 461
2018/02/12 20:26:43 worker.go:99: Worker listening at address: [::]:7080
2018/02/12 20:26:43 pool.go:118: == CONNECT ==> Setting 192.168.206.84:5080
2018/02/12 20:26:43 groups.go:109: Connected to group zero. Connection state: member:<id:461 addr:"192.168.200.88:7080" > state:<counter:879041 groups:<key:1 value:<members:<key:459 value:<id:459 group_id:1 addr:"192.168.206.84:7080" last_update:1518465931 > > members:<key:460 value:<id:460 group_id:1 addr:"192.168.171.220:7080" > > members:<key:461 value:<id:461 group_id:1 addr:"192.168.200.88:7080" leader:true last_update:1518465993 > > tablets:<key:"_predicate_" value:<group_id:1 predicate:"_predicate_" space:56836755 > > tablets:<key:"dummy" value:<group_id:1 predicate:"dummy" space:17 > > tablets:<key:"has_connection" value:<group_id:1 predicate:"has_connection" space:123985764 > > tablets:<key:"name" value:<group_id:1 predicate:"name" space:38151179 > > tablets:<key:"person_id" value:<group_id:1 predicate:"person_id" space:54935649 > > > > zeros:<key:459 value:<id:459 addr:"192.168.206.84:5080" leader:true > > zeros:<key:460 value:<id:460 addr:"192.168.171.220:5080" > > zeros:<key:461 value:<id:461 addr:"192.168.200.88:5080" > > maxLeaseId:290000 maxTxnTs:4880000 >
2018/02/12 20:26:43 pool.go:168: Echo error from 192.168.206.84:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
2018/02/12 20:26:43 pool.go:118: == CONNECT ==> Setting 192.168.206.84:7080
2018/02/12 20:26:43 pool.go:168: Echo error from 192.168.171.220:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
2018/02/12 20:26:43 pool.go:118: == CONNECT ==> Setting 192.168.171.220:7080
2018/02/12 20:26:43 pool.go:118: == CONNECT ==> Setting 192.168.200.88:5080
2018/02/12 20:26:43 pool.go:118: == CONNECT ==> Setting 192.168.171.220:5080
2018/02/12 20:26:43 draft.go:140: Node ID: 461 with GroupID: 1
2018/02/12 20:26:44 node.go:231: Found Snapshot, Metadata: {ConfState:{Nodes:[459 460 461] XXX_unrecognized:[]} Index:559475 Term:38 XXX_unrecognized:[]}
2018/02/12 20:26:44 node.go:246: Found hardstate: {Term:52 Vote:461 Commit:1756177 XXX_unrecognized:[]}
2018/02/12 20:26:53 pool.go:168: Echo error from 192.168.206.84:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
2018/02/12 20:26:53 pool.go:168: Echo error from 192.168.171.220:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
2018/02/12 20:26:53 node.go:258: Group 1 found 1196702 entries
2018/02/12 20:26:53 draft.go:661: Restarting node for group: 1
2018/02/12 20:26:53 raft.go:567: INFO: 1cd became follower at term 52
2018/02/12 20:26:53 raft.go:315: INFO: newRaft 1cd [peers: [1cb,1cc,1cd], term: 52, commit: 1756177, applied: 559475, lastindex: 1756177, lastterm: 38]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x98 pc=0x1000ed6]

goroutine 657 [running]:
github.com/dgraph-io/dgraph/posting.getNew(0xc46bee0760, 0x17, 0x17, 0xc42000e0d0, 0x0, 0x0, 0x0)
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/posting/mvcc.go:470 +0x286
github.com/dgraph-io/dgraph/posting.Get(0xc46bee0760, 0x17, 0x17, 0xc46bee0760)
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/posting/lists.go:253 +0xca
github.com/dgraph-io/dgraph/worker.runMutation(0x1a18000, 0xc46bead800, 0xc467e72180, 0xc467e703c0, 0x124e480, 0xc46bebad20)
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/worker/mutation.go:80 +0x231
github.com/dgraph-io/dgraph/worker.(*node).processMutation(0xc4201a0310, 0xc467e6c640, 0x1, 0xc42031acc0)
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/worker/draft.go:279 +0x154
github.com/dgraph-io/dgraph/worker.(*scheduler).processTasks(0xc45f7001a0)
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/worker/scheduler.go:65 +0x6b
created by github.com/dgraph-io/dgraph/worker.(*scheduler).init
        /home/travis/gopath/src/github.com/dgraph-io/dgraph/worker/scheduler.go:56 +0xe9
```

If it's helpful, here are snippets from the dgraph zero log from one of the three machines:
```
Setting up grpc listener at: 0.0.0.0:5080
Setting up http listener at: 0.0.0.0:6080
2018/02/12 20:25:33 node.go:231: Found Snapshot, Metadata: {ConfState:{Nodes:[459 460 461] XXX_unrecognized:[]} Index:280205 Term:18694 XXX_unrecognized:[]}
2018/02/12 20:25:33 node.go:246: Found hardstate: {Term:18947 Vote:459 Commit:879030 XXX_unrecognized:[]}
2018/02/12 20:25:35 node.go:258: Group 0 found 598825 entries
2018/02/12 20:25:35 raft.go:405: Restarting node for dgraphzero
2018/02/12 20:25:35 pool.go:168: Echo error from 192.168.171.220:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
2018/02/12 20:25:35 pool.go:118: == CONNECT ==> Setting 192.168.171.220:7080
2018/02/12 20:25:35 pool.go:168: Echo error from 192.168.200.88:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
2018/02/12 20:25:35 pool.go:118: == CONNECT ==> Setting 192.168.200.88:7080
2018/02/12 20:25:35 pool.go:168: Echo error from 192.168.206.84:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
2018/02/12 20:25:35 pool.go:118: == CONNECT ==> Setting 192.168.206.84:7080
2018/02/12 20:25:35 pool.go:118: == CONNECT ==> Setting 192.168.171.220:5080
2018/02/12 20:25:35 pool.go:118: == CONNECT ==> Setting 192.168.200.88:5080
2018/02/12 20:25:35 raft.go:567: INFO: 1cb became follower at term 18947
2018/02/12 20:25:35 raft.go:315: INFO: newRaft 1cb [peers: [1cb,1cc,1cd], term: 18947, commit: 879030, applied: 280205, lastindex: 879030, lastterm: 18947]
Running Dgraph zero...
2018/02/12 20:25:35 raft.go:708: INFO: 1cb [term: 18947] received a MsgHeartbeat message with higher term from 1cd [term: 18948]
2018/02/12 20:25:35 raft.go:567: INFO: 1cb became follower at term 18948
2018/02/12 20:25:35 node.go:301: INFO: raft.node: 1cb elected leader 1cd at term 18948
2018/02/12 20:25:35 raft.go:361: Could not apply lease, ignoring: proposedLease={1619314431 <nil> <nil> 0 2100000 0 <nil>} maxLeaseId=260000 maxTxnTs=4680000
2018/02/12 20:25:35 raft.go:361: Could not apply lease, ignoring: proposedLease={46114194 <nil> <nil> 0 2110000 0 <nil>} maxLeaseId=260000 maxTxnTs=4680000
2018/02/12 20:25:35 raft.go:361: Could not apply lease, ignoring: proposedLease={3553001265 <nil> <nil> 0 2120000 0 <nil>} maxLeaseId=260000 maxTxnTs=4680000
2018/02/12 20:25:35 raft.go:361: Could not apply lease, ignoring: proposedLease={3397873571 <nil> <nil> 0 2130000 0 <nil>} maxLeaseId=260000 maxTxnTs=4680000
2018/02/12 20:25:35 raft.go:361: Could not apply lease, ignoring: proposedLease={3926416813 <nil> <nil> 0 2140000 0 <nil>} maxLeaseId=260000 maxTxnTs=4680000
2018/02/12 20:25:35 raft.go:361: Could not apply lease, ignoring: proposedLease={2627986296 <nil> <nil> 0 2150000 0 <nil>} maxLeaseId=260000 maxTxnTs=4680000
2018/02/12 20:25:35 raft.go:361: Could not apply lease, ignoring: proposedLease={3432128265 <nil> <nil> 0 2160000 0 <nil>} maxLeaseId=260000 maxTxnTs=4680000
2018/02/12 20:25:35 raft.go:361: Could not apply lease, ignoring: proposedLease={3617727421 <nil> <nil> 0 2170000 0 <nil>} maxLeaseId=260000 maxTxnTs=4680000
2018/02/12 20:25:35 raft.go:361: Could not apply lease, ignoring: proposedLease={2576411620 <nil> <nil> 0 2180000 0 <nil>} maxLeaseId=260000 maxTxnTs=4680000
2018/02/12 20:25:35 raft.go:361: Could not apply lease, ignoring: proposedLease={144966137 <nil> <nil> 0 2190000 0 <nil>} maxLeaseId=260000 maxTxnTs=4680000
2018/02/12 20:25:35 raft.go:361: Could not apply lease, ignoring: proposedLease={3781064113 <nil> <nil> 0 2200000 0 <nil>} maxLeaseId=260000 maxTxnTs=4680000
2018/02/12 20:25:35 raft.go:361: Could not apply lease, ignoring: proposedLease={3837184439 <nil> <nil> 0 2210000 0 <nil>} maxLeaseId=260000 maxTxnTs=4680000
2018/02/12 20:25:35 raft.go:361: Could not apply lease, ignoring: proposedLease={3686945825 <nil> <nil> 0 2220000 0 <nil>} maxLeaseId=260000 maxTxnTs=4680000
2018/02/12 20:25:35 raft.go:361: Could not apply lease, ignoring: proposedLease={2523158851 <nil> <nil> 0 2230000 0 <nil>} maxLeaseId=260000 maxTxnTs=4680000
2018/02/12 20:25:35 raft.go:361: Could not apply lease, ignoring: proposedLease={2914101075 <nil> <nil> 120000 0 0 <nil>} maxLeaseId=260000 maxTxnTs=4680000
2018/02/12 20:25:35 raft.go:361: Could not apply lease, ignoring: proposedLease={963481398 <nil> <nil> 0 2240000 0 <nil>} maxLeaseId=260000 maxTxnTs=4680000
2018/02/12 20:25:35 raft.go:361: Could not apply lease, ignoring: proposedLease={923131383 <nil> <nil> 0 2250000 0 <nil>} maxLeaseId=260000 maxTxnTs=4680000
2018/02/12 20:25:35 raft.go:361: Could not apply lease, ignoring: proposedLease={3144537441 <nil> <nil> 0 2260000 0 <nil>} maxLeaseId=260000 maxTxnTs=4680000

<snip>

2018/02/12 20:25:37 raft.go:361: Could not apply lease, ignoring: proposedLease={3850574842 <nil> <nil> 0 4670000 0 <nil>} maxLeaseId=260000 maxTxnTs=4680000
2018/02/12 20:25:37 raft.go:361: Could not apply lease, ignoring: proposedLease={4030234637 <nil> <nil> 0 4680000 0 <nil>} maxLeaseId=260000 maxTxnTs=4680000
2018/02/12 20:25:43 oracle.go:375: No healthy connection found to leader of group 1
2018/02/12 20:25:45 raft.go:749: INFO: 1cb is starting a new election at term 18948
2018/02/12 20:25:45 raft.go:580: INFO: 1cb became candidate at term 18949
2018/02/12 20:25:45 raft.go:664: INFO: 1cb received MsgVoteResp from 1cb at term 18949
2018/02/12 20:25:45 raft.go:651: INFO: 1cb [logterm: 18948, index: 879036] sent MsgVote request to 1cc at term 18949
2018/02/12 20:25:45 raft.go:651: INFO: 1cb [logterm: 18948, index: 879036] sent MsgVote request to 1cd at term 18949
2018/02/12 20:25:45 node.go:307: INFO: raft.node: 1cb lost leader 1cd at term 18949
2018/02/12 20:25:45 raft.go:664: INFO: 1cb received MsgVoteResp from 1cc at term 18949
2018/02/12 20:25:45 raft.go:1013: INFO: 1cb [quorum:2] has received 2 MsgVoteResp votes and 0 vote rejections
2018/02/12 20:25:45 raft.go:621: INFO: 1cb became leader at term 18949
2018/02/12 20:25:45 node.go:301: INFO: raft.node: 1cb elected leader 1cb at term 18949
2018/02/12 20:25:45 node.go:343: Error while sending message to node with addr: 192.168.200.88:5080, err: rpc error: code = Unknown desc = No node has been set up yet
2018/02/12 20:25:45 node.go:343: Error while sending message to node with addr: 192.168.200.88:5080, err: rpc error: code = Unknown desc = No node has been set up yet
2018/02/12 20:25:45 node.go:343: Error while sending message to node with addr: 192.168.200.88:5080, err: rpc error: code = Unknown desc = No node has been set up yet
2018/02/12 20:25:45 node.go:343: Error while sending message to node with addr: 192.168.200.88:5080, err: rpc error: code = Unknown desc = No node has been set up yet
2018/02/12 20:25:45 pool.go:168: Echo error from 192.168.171.220:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
2018/02/12 20:25:45 node.go:343: Error while sending message to node with addr: 192.168.200.88:5080, err: rpc error: code = Unknown desc = No node has been set up yet
2018/02/12 20:25:45 pool.go:168: Echo error from 192.168.206.84:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
2018/02/12 20:25:45 pool.go:168: Echo error from 192.168.200.88:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure

<snip>

2018/02/12 20:25:47 node.go:343: Error while sending message to node with addr: 192.168.200.88:5080, err: rpc error: code = Unknown desc = No node has been set up yet
2018/02/12 20:25:53 oracle.go:375: No healthy connection found to leader of group 1
2018/02/12 20:25:55 pool.go:168: Echo error from 192.168.171.220:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
2018/02/12 20:25:55 pool.go:168: Echo error from 192.168.200.88:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
2018/02/12 20:25:55 pool.go:168: Echo error from 192.168.206.84:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
2018/02/12 20:26:03 oracle.go:375: No healthy connection found to leader of group 1
2018/02/12 20:26:05 pool.go:168: Echo error from 192.168.171.220:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
2018/02/12 20:26:05 pool.go:168: Echo error from 192.168.200.88:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
2018/02/12 20:26:05 pool.go:168: Echo error from 192.168.206.84:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
2018/02/12 20:26:11 zero.go:323: Got connection request: id:459 addr:"192.168.206.84:7080"
2018/02/12 20:26:11 zero.go:420: Connected
2018/02/12 20:26:13 oracle.go:375: No healthy connection found to leader of group 1
2018/02/12 20:26:15 pool.go:168: Echo error from 192.168.171.220:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
2018/02/12 20:26:15 pool.go:168: Echo error from 192.168.200.88:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
2018/02/12 20:26:15 pool.go:168: Echo error from 192.168.206.84:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
2018/02/12 20:26:23 oracle.go:375: No healthy connection found to leader of group 1
2018/02/12 20:26:25 pool.go:168: Echo error from 192.168.171.220:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
2018/02/12 20:26:25 pool.go:168: Echo error from 192.168.200.88:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
2018/02/12 20:26:25 pool.go:168: Echo error from 192.168.206.84:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
2018/02/12 20:26:33 oracle.go:375: No healthy connection found to leader of group 1
2018/02/12 20:26:35 pool.go:168: Echo error from 192.168.171.220:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
2018/02/12 20:26:35 pool.go:168: Echo error from 192.168.200.88:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
2018/02/12 20:26:35 pool.go:168: Echo error from 192.168.206.84:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
2018/02/12 20:26:43 zero.go:323: Got connection request: id:461 addr:"192.168.200.88:7080"
2018/02/12 20:26:43 zero.go:420: Connected

<snip>

2018/02/12 20:35:23 oracle.go:375: No healthy connection found to leader of group 1
2018/02/12 20:35:25 pool.go:168: Echo error from 192.168.171.220:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
2018/02/12 20:35:25 pool.go:168: Echo error from 192.168.200.88:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
2018/02/12 20:35:25 pool.go:168: Echo error from 192.168.206.84:7080. Err: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
```

**Comment by pawanrawal** [comment]
> signal SIGSEGV: segmentation violation code=0x1 addr=0x98 pc=0x1000ed6]
> 
> goroutine 657 [running]:
> github.com/dgraph-io/dgraph/posting.getNew(0xc46bee0760, 0x17, 0x17, 0xc42000e0d0, 0x0, 0x0, 0x0)
>         /home/travis/gopath/src/github.com/dgraph-io/dgraph/posting/mvcc.go:470 +0x286

This was fixed by 2c63353d8517922366f6ca909cba2a367df6717d.

> 2018/02/12 20:27:33 Assert failed
> github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger/y.AssertTrue
>         /home/travis/gopath/src/github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger/y/error.go:55
> github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger.replayFunction.func2
>         /home/travis/gopath/src/github.com/dgraph-io/dgraph/vendor/github.com/dgraph-io/badger/db.go:138

This was fixed by https://github.com/dgraph-io/badger/commit/b5727f1b8485a1812e3e904216cc01de7346dce5. I will close this issue when badger v2.0 is vendored in Dgraph.

**Comment by pawanrawal** [comment]
We have vendored in an updated version of badger with ab9846a7b9091e2e1fa97cfb33fad5732a2d8493 which has the fix for this issue.
